NDTensors/ext/NDTensorsCUDAExt/linearalgebra.jl

```diff
@@ -1,13 +1,39 @@
-function NDTensors.svd_catch_error(A::CuMatrix; alg="JacobiAlgorithm")
+function NDTensors.svd_catch_error(A::CuMatrix; alg::String="JacobiAlgorithm")
```

Comment:
We should rename this to `"jacobi_algorithm"`, following the convention that keyword arguments should be lowercase.

Reply:
Also, we should explicitly check for an input of `"qr_algorithm"`, and then throw an error if the `alg` is something besides `"jacobi_algorithm"` and `"qr_algorithm"`.

Reply:
I changed it to lowercase. I was thinking since SVD in NDTensors has a way to [catch incorrect keywords](https://github.com/ITensor/ITensors.jl/blob/e1c493d0b0a28148f99ddad30627e74db7322926/NDTensors/src/linearalgebra/linearalgebra.jl#L108C1-L136C6) it wouldn't be necessary here but I will add it anyway

Reply:
Looks like throwing an error also causes an issue for tests when we are trying to call CPU svd backends with CU. Question for you (especially in the tests). Should I catch this problem in the code and just run the faster CUDA `qr_algorithm` or in say `NDTensors/tests` should I do something like 
```
if dev == NDTensors.cu
  svd(A; alg = "qr_algorith")
else 
  svd(A; alg="divide_and_conquer")
end
```

Reply:
@mtfishman I could also make `default_svd_alg(a, ::AbstractArray) = divide_and_conquer` and in `NDTensorsCUDA` make `default_svd_alg(a, ::CuArray) = qr_algorithm` and then just change the line `default_svd_alg(a, unwrap_type(a))`?

Reply:
I think we should change `default_svd_alg`, since we should pick the best option so users don't have to pick it themselves. Could use `expose` here.

Reply:
Great I have a version coded and am running test sto verify that it works properly